tests/e2e/basic/_basic_test_setup/setup__e2e_resources.py

```diff
@@ -0,0 +1,51 @@
+import pytest
+
+
+@pytest.mark.sanity
+@pytest.mark.bridge
+@pytest.mark.nat
+@pytest.mark.vlan
+@pytest.mark.firmware
+class TestFirmware(object):
+
+    @pytest.mark.firmware_create
+    def test_firmware_create(self, upload_firmware, instantiate_testrail, instantiate_project, test_cases):
+        if upload_firmware != 0:
+            instantiate_testrail.update_testrail(case_id=test_cases["create_fw"], run_id=instantiate_project,
+                                                 status_id=1,
+                                                 msg='Create new FW version by API successful')
+            PASS = True
+        else:
+            instantiate_testrail.update_testrail(case_id=test_cases["create_fw"], run_id=instantiate_project,
```

Comment:
Can't we move the upgrade_testrail to the bottom and pass in value of status and msg from the if/else condition?